digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/dataimport/CsvDataImporter.scala

```diff
@@ -157,21 +213,96 @@ class TrafficSignCsvImporter extends CsvDataImporterOperations {
+    trafficSignAttributes.properties.find (prop => prop.columnName == propertyName).map(_.value)
+  }
+
+  private def generateBaseProperties(trafficSignAttributes: CsvAssetRow) : Set[SimpleTrafficSignProperty] = {
+    val valueProperty = tryToInt(getPropertyValue(trafficSignAttributes, "value").toString).map { value =>
+      SimpleTrafficSignProperty(valuePublicId, Seq(TextPropertyValue(value.toString)))}
+
+    val additionalInfo = getPropertyValue(trafficSignAttributes, "additionalInfo").toString
+    val additionalProperty = if(additionalInfo.nonEmpty)
+        Some(SimpleTrafficSignProperty(infoPublicId, Seq(TextPropertyValue(additionalInfo))))
+      else
+        None
+
+    val typeProperty = SimpleTrafficSignProperty(typePublicId, Seq(TextPropertyValue(TrafficSignType.applyTRValue(getPropertyValue(trafficSignAttributes, "trafficSignType").toString.toInt).OTHvalue.toString)))
+
+    Set(Some(typeProperty), valueProperty, additionalProperty).flatten
+  }
+
+  def createTrafficSigns(trafficSignAttributes: Seq[CsvAssetRowAndRoadLink]): Seq[AdditionalPanelInfo] = {
+
+    val signs = trafficSignAttributes.map { trafficSignAttribute =>
+      val properties = trafficSignAttribute.properties
+      val nearbyLinks = trafficSignAttribute.roadLink
+      val enrichedNearbyLinks = trafficSignAttribute.enrichedRoadLink
+      val optBearing = tryToInt(getPropertyValue(properties, "bearing").toString)
+      val twoSided = getPropertyValue(properties, "twoSided") match {
+        case "1" => true
+        case _ => false
+      }
+      val lon = getPropertyValue(properties, "lon").asInstanceOf[BigDecimal].toLong
+      val lat = getPropertyValue(properties, "lat").asInstanceOf[BigDecimal].toLong
+
+      if(enrichedNearbyLinks.nonEmpty) {
+        val closestRoadLink = Seq(enrichedNearbyLinks.minBy(r => GeometryUtils.minimumDistance(Point(lon.toLong, lat.toLong), r.geometry))).head
+        val validityDirection = trafficSignService.getValidityDirection(Point(lon, lat), closestRoadLink, optBearing, twoSided)
+
+        val mValue = GeometryUtils.calculateLinearReferenceFromPoint(Point(lon, lat), closestRoadLink.geometry)
+        val bearingRecalculation = GeometryUtils.calculateBearing(closestRoadLink.geometry, Some(mValue), Some(closestRoadLink.length) )
+
+        CsvTrafficSign(lon, lat, closestRoadLink.linkId, generateBaseProperties(properties), validityDirection, Some(bearingRecalculation), mValue, Some(closestRoadLink), nearbyLinks)
+      } else {
+        val closestRoadLink = Seq(nearbyLinks.minBy(r => GeometryUtils.minimumDistance(Point(lon.toLong, lat.toLong), r.geometry))).head
+        val trafficSignDirection = trafficSignService.getTrafficSignValidityDirection(Point(lon, lat), closestRoadLink.geometry)
+
+        CsvTrafficSign(lon, lat, 0, generateBaseProperties(properties), trafficSignDirection, None, 0, None, Seq(closestRoadLink))
+      }
```

Comment:
please check this conditions, 
Doesn't make sense create a CSVTrafficSign without linkId, the enrichedNearbyLinks and nearbyLinks should be the same. If we need work with enrichedRoadLinks, lets replace and remove nearbyLinks